example_2/hardware/diffbot_system.cpp

```diff
@@ -178,18 +171,22 @@ hardware_interface::return_type DiffBotSystemHardware::read(
   // BEGIN: This part here is for exemplary purposes - Please do not copy to your production code
   std::stringstream ss;
   ss << "Reading states:";
-  for (std::size_t i = 0; i < hw_velocities_.size(); i++)
+  ss << std::fixed << std::setprecision(2);
+  for (const auto & [name, descr] : joint_state_interfaces_)
   {
-    // Simulate DiffBot wheels's movement as a first-order system
-    // Update the joint status: this is a revolute joint without any limit.
-    // Simply integrates
-    hw_positions_[i] = hw_positions_[i] + period.seconds() * hw_velocities_[i];
-
-    ss << std::fixed << std::setprecision(2) << std::endl
-       << "\t"
-          "position "
-       << hw_positions_[i] << " and velocity " << hw_velocities_[i] << " for '"
-       << info_.joints[i].name.c_str() << "'!";
+    if (descr.get_interface_name() == hardware_interface::HW_IF_POSITION)
+    {
+      // Simulate DiffBot wheels's movement as a first-order system
+      // Update the joint status: this is a revolute joint without any limit.
+      // Simply integrates
+      auto velo = get_command(descr.get_prefix_name() + "/" + hardware_interface::HW_IF_VELOCITY);
+      set_state(name, get_state(name) + period.seconds() * velo);
+
+      ss << std::endl
+         << "\t"
+            "position "
+         << get_state(name) << " and velocity " << velo << " for '" << name << "'!";
```

Comment:
```suggestion
      ss << std::endl  << "\t position "
           << get_state(name) << " and velocity " << velo << " for '" << name << "'!";
```

Reply:
still looks better :D one thing none of these formatters do is to join strings back together. Seeing a string split into multiple lines always triggers me